frontend/packages/ceph-storage-plugin/src/components/independent-mode/install.tsx

```diff
@@ -0,0 +1,266 @@
+);
+
+const getErrorText = (text: string) => <ErrorText text={text} />;
+
+const InstallExternalCluster = withHandlePromise((props: InstallExternalClusterProps) => {
+  const {
+    inProgress,
+    errorMessage,
+    handlePromise,
+    match: {
+      params: { ns, appName },
+    },
+  } = props;
+  const [clusterServiceVersion, setClusterServiceVersion] = React.useState(null);
+  const [fileData, setFileData] = React.useState('');
+  const [fileName, setFileName] = React.useState('');
+  const [namespace, setNamespace] = React.useState(ns);
+  const [externalFSID, setExternalFSID] = React.useState(null);
+  const [externalAdminSecret, setExternalAdminSecret] = React.useState(null);
+  const [externalMonData, setExternalMonData] = React.useState(null);
+  const [dataError, setDataError] = React.useState(ERROR);
+  const [fileError, setFileError] = React.useState('');
+  const [, updateState] = React.useState();
+
+  // Todo(bipuladh): React does shallow comparison dataError and fileError need deep comparison.
+  const forceUpdate = React.useCallback(() => updateState({}), []);
+
+  const setErrors = React.useCallback((errors: ErrorType[]): void => {
+    for (const err of errors) {
+      setDataError(Object.assign(ERROR, { [err.field]: err.message }));
+    }
+  }, []);
+
+  const getState = React.useCallback(
+    () => ({
+      [Field.NAMESPACE]: namespace,
+      [Field.FSID]: externalFSID,
+      [Field.ADMIN]: externalAdminSecret,
+      [Field.MONDATA]: externalMonData,
+    }),
+    [externalAdminSecret, externalFSID, externalMonData, namespace],
+  );
+
+  React.useEffect(() => {
+    setErrors(checkError(getState()));
+  }, [getState, setErrors]);
+
+  // File Upload handler
+  const onUpload = (event) => {
+    event.preventDefault();
+    const reader = new FileReader();
+    const file = event.target.files[0];
+    setFileName(file.name);
+    reader.onload = (ev) => {
+      const data = ev.target?.result as string;
+      setFileData(data);
+    };
+    reader.readAsText(file);
+  };
+
+  const onSubmit = (event) => {
+    event.preventDefault();
+    // https://github.com/rook/rook/blob/master/cluster/examples/kubernetes/ceph/import-external-cluster.sh
+    const promises = [];
+    const secret = {
+      apiVersion: SecretModel.apiVersion,
+      kind: SecretModel.kind,
+      metadata: {
+        name: 'rook-ceph-mon',
+        namespace,
+      },
+      stringData: {
+        'cluster-name': namespace,
+        fsid: externalFSID,
+        'admin-secret': externalAdminSecret,
+        'mon-secret': externalMonData,
+      },
+      type: 'Opaque',
+    };
+    const cmap = {
+      apiVersion: ConfigMapModel.apiVersion,
+      kind: ConfigMapModel.kind,
+      metadata: {
+        name: 'rook-ceph-mon-endpoints',
+        namespace,
+      },
+      data: {
+        data: `${externalMonData}`,
+        mapping: '{}',
+        maxMonId: '2',
+      },
+    };
+    const ocsObj = {
+      apiVersion: 'ocs.openshift.io/v1',
```

Comment:
```suggestion
      apiVersion: OCSServiceModel.apiVersion',
```

Reply:
that has a value of v1 and for some weird reason, the API call dint go through. So hardcoded it to that.

Reply:
Yes, I can also see that, for all other we are using apiVersion. We probably need to check with OCS team.

Reply:
Yup weird.